### PR TITLE
Wasm components

### PR DESCRIPTION
**Pull request by tkeburia**

### CODE REVIEW

File: wasm-components/rust/http-controller/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            line_items: order
+                .line_items
+                .iter()
+                .map(|line_item| LineItemData::from(line_item.clone()))
+                .collect(),
+        }
+    }
+}
+
+impl Guest for HttpServer {
```
Comment (charypar) [inline wasm-components/rust/http-controller/src/lib.rs:94]: This WASI glue for HTTP really needs to become a library... makes my eyes hurt what you need to do to handle a simple request 🙈

File: wasm-components/rust/notification-service/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        crate::loud_print!(order_notification.order_number);
+
+        Ok(())
+    }
+}
+
+export!(HttpServer);
+
+#[macro_export]
+macro_rules! loud_print {
```
Comment (charypar) [inline wasm-components/rust/notification-service/src/lib.rs:26]: 📢

File: wasm-components/rust/inventory-service/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+use common::inventory::Availability as AvailabilityData;
+use exports::platform_poc::inventory::inventory::Guest;
+use platform_poc::inventory::types::{Availability, Error};
+use wasmcloud::postgres::query::query;
+use wasmcloud::postgres::query::PgValue;
+use wasmcloud::postgres::types::ResultRowEntry;
+
+use wasi::logging::logging::{log, Level};
+
+struct HttpServer;
```
Comment (charypar) [inline wasm-components/rust/inventory-service/src/lib.rs:14]: Is this meant to be called `InventoryService`?

File: wasm-components/rust/inventory-service/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,70 @@
+wit_bindgen::generate!({
+    world: "inventory-service"
+});
+
+use common::inventory::Availability as AvailabilityData;
+use exports::platform_poc::inventory::inventory::Guest;
```
Comment (charypar) [inline wasm-components/rust/inventory-service/src/lib.rs:6]: Worth renaming these imports to the interface name, e.g. `as Inventory` or something similar.

**Review by charypar — APPROVED** [review]
This is very cool!

Are you planning to do one that instead composes all the components into one big one and deploys it as a (absolute) unit?

**Comment by tkeburia** [comment]
> This is very cool!
> 
> Are you planning to do one that instead composes all the components into one big one and deploys it as a (absolute) unit?

@charypar definitely on my list. But first I want to provision a wasmcloud host somewhere we can all deploy to - e.g. k8s cluster on Civo

File: wasm-components/rust/http-controller/src/lib.rs (hunk 1 above)
Comment (tkeburia) [reply wasm-components/rust/http-controller/src/lib.rs:94]: Lol yeah - I don't want to admit how long it took me to figure out how to read a body off an http request 🙈

**Pull request merged**